lib/Gateway.js

```diff
@@ -786,6 +807,77 @@ function addRgbDimmer (cfg, node, valueId) {
   }
 
   return false
+} */
+
+function addRgbColorSwitch (cfg, node, valueId) {
+  cfg = copy(cfg)
+  cfg.discovery_payload.rgb_state_topic = this.mqtt.getTopic(
+    this.valueTopic(node, valueId)
+  )
+  cfg.discovery_payload.rgb_command_topic =
+    cfg.discovery_payload.rgb_state_topic + '/set'
+
+  // The following part of code, checks if ML or Binary works. If one exists the other
+  let brightnessValue = false
+  let switchValue = false
+  if (node.values.hasOwnProperty(`38-${valueId.endpoint}-currentValue`)) {
+    brightnessValue = `38-${valueId.endpoint}-currentValue`
+  } else if (
+    node.values.hasOwnProperty(`38-${valueId.endpoint + 1}-currentValue`)
+  ) {
+    brightnessValue = `38-${valueId.endpoint + 1}-currentValue`
+  } else if (
+    node.values.hasOwnProperty(`37-${valueId.endpoint}-currentValue`)
+  ) {
+    switchValue = `37-${valueId.endpoint}-currentValue`
+  }
+
+  /* Find the control switch of the device Brightness or Binary
+     If multilevel is not there use binary
+     Some devices use also endpoint + 1 as on/off/brightness... try to guess that too!
+  */
+  if (brightnessValue) {
+    cfg.discovery_payload.brightness_state_topic = this.mqtt.getTopic(
+      this.valueTopic(node, node.values[brightnessValue])
+    )
+    cfg.discovery_payload.brightness_command_topic = this.mqtt.getTopic(
+      this.valueTopic(node, node.values[brightnessValue], true).targetTopic
+    ) + '/set'
+    cfg.discovery_payload.state_topic = false
+  } else if (switchValue) {
+    cfg.discovery_payload.state_topic = this.mqtt.getTopic(
+      this.valueTopic(node, node.values[brightnessValue])
+    )
+    cfg.discovery_payload.command_topic = this.mqtt.getTopic(
+      this.valueTopic(node, node.values[brightnessValue], true).targetTopic
```

Comment:
use a var here as you are using the same value in the previous step

Reply:
refactored it to use a var